src/drivers/github.go

```diff
@@ -1,26 +1,164 @@
 package drivers
 
 import (
+	"context"
+	"errors"
 	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"golang.org/x/oauth2"
 
 	"github.com/Originate/git-town/src/git"
+	"github.com/google/go-github/github"
 )
 
 // GithubCodeHostingDriver provides tools for working with repositories
 // hosted on Github
-type GithubCodeHostingDriver struct{}
+type GithubCodeHostingDriver struct {
+	client     *github.Client
+	owner      string
+	repository string
+}
+
+// NewGithubCodeHostingDriver returns a new GithubCodeHostingDriver instance
+func NewGithubCodeHostingDriver(fullRepository string) *GithubCodeHostingDriver {
```

Comment:
Please note that #976 injects the repo URL via a `SetOriginURL` method now, no longer using the constructor.